pkgs/by-name/ta/tabby/package.nix

```diff
@@ -107,55 +131,85 @@ rustPlatform.buildRustPackage {
+    rev = "refs/tags/v${version}";
+    hash = "sha256-8clEBWAT+HI2eecOsmldgRcA58Ehq9bZT4ZwUMm494g=";
     fetchSubmodules = true;
   };
 
   cargoLock = {
     lockFile = ./Cargo.lock;
     outputHashes = {
-      "apalis-0.5.1" = "sha256-hGvVuSy32lSTR5DJdiyf8q1sXbIeuLSGrtyq6m2QlUQ=";
-      "tree-sitter-c-0.20.6" = "sha256-Etl4s29YSOxiqPo4Z49N6zIYqNpIsdk/Qd0jR8jdvW4=";
-      "tree-sitter-cpp-0.20.3" = "sha256-UrQ48CoUMSHmlHzOMu22c9N4hxJtHL2ZYRabYjf5byA=";
-      "tree-sitter-solidity-0.0.3" = "sha256-b+LthCf+g19sjKeNgXZmUV0RNi94O3u0WmXfgKRpaE0=";
+      "ollama-rs-0.1.9" = "sha256-d6sKUxc8VQbRkVqMOeNFqDdKesq5k32AQShK67y2ssg=";
+      "oneshot-0.1.6" = "sha256-PmYuHuNTqToMyMHPRFDUaHUvFkVftx9ZCOBwXj+4Hc4=";
+      "ownedbytes-0.7.0" = "sha256-p0+ohtW0VLmfDTZw/LfwX2gYfuYuoOBcE+JsguK7Wn8=";
+      "sqlx-0.7.4" = "sha256-tcISzoSfOZ0jjNgGpuPPxjMxmBUPw/5FVDoALZEAHKY=";
+      "tree-sitter-c-0.21.3" = "sha256-ucbHLS2xyGo1uyKZv/K1HNXuMo4GpTY327cgdVS9F3c=";
+      "tree-sitter-cpp-0.22.1" = "sha256-3akSuQltFMF6I32HwRU08+Hcl9ojxPGk2ZuOX3gAObw=";
+      "tree-sitter-solidity-1.2.6" = "sha256-S00hdzMoIccPYBEvE092/RIMnG8YEnDGk6GJhXlr4ng=";
     };
   };
 
   # https://github.com/TabbyML/tabby/blob/v0.7.0/.github/workflows/release.yml#L39
-  cargoBuildFlags = [
-    "--release"
-    "--package" "tabby"
-  ] ++ optionals enableRocm [
-    "--features" "rocm"
-  ] ++ optionals enableCuda [
-    "--features" "cuda"
+  cargoBuildFlags =
+    [
+      # Don't need to build llama-cpp-server (included in default build)
+      "--no-default-features"
+      "--features"
+      "ee"
+      "--package"
+      "tabby"
+    ]
+    ++ optionals enableRocm [
+      "--features"
+      "rocm"
+    ]
+    ++ optionals enableCuda [
+      "--features"
+      "cuda"
+    ];
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
   ];
-
-  OPENSSL_NO_VENDOR = 1;
-
-  nativeBuildInputs = [
-    pkg-config
-    protobuf
-    git
-  ] ++ optionals enableCuda [
-    autoAddDriverRunpath
-  ];
-
-  buildInputs = [ openssl ]
-  ++ optionals stdenv.hostPlatform.isDarwin darwinBuildInputs
-  ++ optionals enableCuda cudaBuildInputs
-  ++ optionals enableRocm rocmBuildInputs
-  ;
-
-  env.LLAMA_CPP_LIB = "${lib.getLib llamaccpPackage}/lib";
-  patches = [ ./0001-nix-build-use-nix-native-llama-cpp-package.patch ];
+  versionCheckProgramArg = [ "--version" ];
+  doInstallCheck = true;
+
+  nativeBuildInputs =
+    [
+      git
+      pkg-config
+      protobuf
+    ]
+    ++ optionals enableCuda [
+      autoAddDriverRunpath
+    ];
+
+  buildInputs =
+    [ openssl ]
+    ++ optionals stdenv.hostPlatform.isDarwin darwinBuildInputs
+    ++ optionals enableCuda cudaBuildInputs
+    ++ optionals enableRocm rocmBuildInputs;
+
+  postInstall = ''
+      # NOTE: Project contains a subproject for building llama-server
+      # But, we already have a derivation for this
+    ln -s ${lib.getExe' llama-cpp "llama-server"} $out/bin/llama-server
+  '';
```

Comment:
My only doubt is about this approach.
Any thought @SuperSandro2000 ?

Reply:
I can try seeing if just wrapping it works. I went straight to this because the project builds the exe and places it next to it and when it can't find the binary it specifically said to be in the same directory with it. I wasn't sure if it was coded to check a relative file path.

Reply:
We can also modify the code directly but does not look that horrible.